src/include/cpu/rpp_cpu_simd.hpp

```diff
@@ -10,45 +10,397 @@
+    _mm_storeu_si128((__m128i *)dstPtrTempG, px1);    /* store [G01|G02|G03|G04|G05|G06|G07|G08|G09|G10|G11|G12|G13|G14|G15|G16] */ \
+    _mm_storeu_si128((__m128i *)dstPtrTempB, px2);    /* store [B01|B02|B03|B04|B05|B06|B07|B08|B09|B10|B11|B12|B13|B14|B15|B16] */ \
+}
+
+#define RPP_LOAD48_I8PLN3_TO_F32PLN3(srcPtrTempR, srcPtrTempG, srcPtrTempB) \
+{ \
+    px0 = _mm_loadu_si128((__m128i *)srcPtrTempR);    /* load [R01|R02|R03|R04|R05|R06|R07|R08|R09|R10|R11|R12|R13|R14|R15|R16] */ \
+    px1 = _mm_loadu_si128((__m128i *)srcPtrTempG);    /* load [G01|G02|G03|G04|G05|G06|G07|G08|G09|G10|G11|G12|G13|G14|G15|G16] */ \
+    px2 = _mm_loadu_si128((__m128i *)srcPtrTempB);    /* load [B01|B02|B03|B04|B05|B06|B07|B08|B09|B10|B11|B12|B13|B14|B15|B16] */ \
+    px0 = _mm_add_epi8(px0, pxConvertI8);    /* convert to u8 for px0 compute */ \
+    px1 = _mm_add_epi8(px1, pxConvertI8);    /* convert to u8 for px1 compute */ \
+    px2 = _mm_add_epi8(px2, pxConvertI8);    /* convert to u8 for px2 compute */ \
+    px3 = _mm_unpackhi_epi8(px0, zero);    /* unpack 8 hi-pixels of px0 */ \
+    px4 = _mm_unpackhi_epi8(px1, zero);    /* unpack 8 hi-pixels of px1 */ \
+    px5 = _mm_unpackhi_epi8(px2, zero);    /* unpack 8 hi-pixels of px2 */ \
+    px0 = _mm_unpacklo_epi8(px0, zero);    /* unpack 8 lo-pixels of px0 */ \
+    px1 = _mm_unpacklo_epi8(px1, zero);    /* unpack 8 lo-pixels of px1 */ \
+    px2 = _mm_unpacklo_epi8(px2, zero);    /* unpack 8 lo-pixels of px2 */ \
+    p0R = _mm_cvtepi32_ps(_mm_unpacklo_epi16(px0, zero));    /* pixels 0-3 of original px0 containing 16 R values */ \
+    p1R = _mm_cvtepi32_ps(_mm_unpackhi_epi16(px0, zero));    /* pixels 4-7 of original px0 containing 16 R values */ \
+    p2R = _mm_cvtepi32_ps(_mm_unpacklo_epi16(px3, zero));    /* pixels 8-11 of original px0 containing 16 R values */ \
+    p3R = _mm_cvtepi32_ps(_mm_unpackhi_epi16(px3, zero));    /* pixels 12-15 of original px0 containing 16 R values */ \
+    p0G = _mm_cvtepi32_ps(_mm_unpacklo_epi16(px1, zero));    /* pixels 0-3 of original px1 containing 16 G values */ \
+    p1G = _mm_cvtepi32_ps(_mm_unpackhi_epi16(px1, zero));    /* pixels 4-7 of original px1 containing 16 G values */ \
+    p2G = _mm_cvtepi32_ps(_mm_unpacklo_epi16(px4, zero));    /* pixels 8-11 of original px1 containing 16 G values */ \
+    p3G = _mm_cvtepi32_ps(_mm_unpackhi_epi16(px4, zero));    /* pixels 12-15 of original px1 containing 16 G values */ \
+    p0B = _mm_cvtepi32_ps(_mm_unpacklo_epi16(px2, zero));    /* pixels 0-3 of original px1 containing 16 G values */ \
+    p1B = _mm_cvtepi32_ps(_mm_unpackhi_epi16(px2, zero));    /* pixels 4-7 of original px1 containing 16 G values */ \
+    p2B = _mm_cvtepi32_ps(_mm_unpacklo_epi16(px5, zero));    /* pixels 8-11 of original px1 containing 16 G values */ \
+    p3B = _mm_cvtepi32_ps(_mm_unpackhi_epi16(px5, zero));    /* pixels 12-15 of original px1 containing 16 G values */ \
+}
+
+#define RPP_STORE48_F32PLN3_TO_I8PKD3(dstPtrTemp) \
```

Comment:
it is bad practice to use macros which assumes parameters from calling routine

Reply:
Done